astropy/conftest.py

```diff
@@ -23,11 +29,25 @@
     # installed. This can be removed once pyopenssl 1.7.20+ is released.
     modules_to_ignore_on_import=['requests'])
 
-try:
-    import matplotlib
-except ImportError:
-    pass
-else:
+if HAS_MATPLOTLIB:
     matplotlib.use('Agg')
 
+matplotlibrc_cache = {}
+
+
+def pytest_configure(config):
+    # do not assign to matplotlibrc_cache in function scope
+    if HAS_MATPLOTLIB:
+        matplotlibrc_cache.update(matplotlib.rcParams)
+        matplotlib.rcdefaults()
+
+
+def pytest_unconfigure(config):
+    # do not assign to matplotlibrc_cache in function scope
+    if HAS_MATPLOTLIB:
+        for k, v in matplotlibrc_cache.items():
+            matplotlib.rcParams[k] = v
```

Comment:
Just to check, does using ``matplotlib.rcParams.update`` not work here instead of the loop?

Reply:
@kyleaoman - It's up to personal preference I think. The main policy we have is to squash when there're overwhelmingly too many commits, or interim commits with refactored code, larger data  for the same PR. 
I prefer squashing when something is too experimental and doesn't work or changes are only trivial PEP8 ones. Here both are a valid solution, so I think it's perfectly fine to have a new commit.